core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -149,26 +183,71 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     String codecName =
         metadata.property(
             TableProperties.METADATA_COMPRESSION, TableProperties.METADATA_COMPRESSION_DEFAULT);
+    // TODO:This is not compatible with the scenario where the user modifies the metadata file
+    // compression codec arbitrarily.
+    // We can inform the user about this bug first, and fix it later.(Do not modify the compressed
+    // format after the table is created.)
     TableMetadataParser.Codec codec = TableMetadataParser.Codec.fromName(codecName);
     String fileExtension = TableMetadataParser.getFileExtension(codec);
-    Path tempMetadataFile = metadataPath(UUID.randomUUID().toString() + fileExtension);
```

Comment:
This change seems unrelated to the PR

Reply:
The IDE automatically optimises this part of the code. In fact, the change in the regular expression above is also due to the IDE's automatic optimisation.

Reply:
As I mentioned above, do not introduce unrelated changes to the pr. If you want to make optimizations like this you need another PR, especially if you are just implementing automated suggestions from static analysis or the IDE